tools/c7n_gcp/c7n_gcp/query.py

```diff
@@ -161,13 +167,57 @@ def resources(self, query=None):
                     local_session(self.session_factory).get_default_project())
                 return []
             raise
-        self._cache.save(key, resources)
-        return self.filter_resources(resources)
 
     def augment(self, resources):
         return resources
 
 
+class ChildResourceManager(QueryResourceManager):
+
+    @classmethod
+    def get_parent_injection_key(cls):
+        parent_resource_type = cls.resource_type.parent[0]
+        return 'c7n:{}'.format(parent_resource_type)
+
+    def get_resource(self, resource_info):
+        parent_resource_info = self._get_parent_resource_info(resource_info)
+        parent_resource_type = self.resource_type.parent[0]
+        resource_manager = self.get_resource_manager(parent_resource_type)
+        instance = resource_manager.get_resource(parent_resource_info)
+        resource_info[self.get_parent_injection_key()] = instance
+
+        return super(ChildResourceManager, self).get_resource(resource_info)
+
+    def _fetch_resources(self, query):
+        if not query:
+            query = {}
+
+        resources = []
+
+        parent_description = self.resource_type.parent
+        parent_resource_type, parent_field, arg_name = parent_description
+        injection_key = self.get_parent_injection_key()
```

Comment:
the term we use for this everywhere else in the code base is 'annotation_key' please rename for consistency.

Reply:
done